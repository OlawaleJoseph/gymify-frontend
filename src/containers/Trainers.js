import React, { useEffect, useState } from 'react';
import Trainer from '../components/TrainerCard';
import { gettRequest } from '../utils/api';
import { trainersUrl } from '../utils/constants';

const Trainers = () => {
  const [trainers, setTrainers] = useState([]);
  useEffect(() => {
    const getTrainers = async () => {
      const { data } = await gettRequest(trainersUrl);
      setTrainers(data);
    };
    getTrainers();
  }, []);

  return (
    <div className="row justify-content-center align-items-center m-0">
      {trainers && trainers.map(({
        id, img_url, first_name, last_name, username, speciality,
      }) => (
        <Trainer
          key={id}
          id={id}
          imgUrl={img_url}
          firstName={first_name}
          lastName={last_name}
          username={username}
          speciality={speciality}
        />
      ))}
    </div>
  );
};

export default Trainers;
