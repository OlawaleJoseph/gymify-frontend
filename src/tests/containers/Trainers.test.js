import Trainers from '../../containers/Trainers';
import { mountRender } from '../setup';

describe('Side Bar', () => {
  const page = mountRender(Trainers);

  test('should render trainers', () => {
    expect(page).toBeTruthy();
    expect(page).toMatchSnapshot();
  });
});
