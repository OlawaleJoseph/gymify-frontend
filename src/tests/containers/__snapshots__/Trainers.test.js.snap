// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Bar should render trainers 1`] = `ReactWrapper {}`;
